src/general/interpolation.jl

```diff
@@ -565,42 +570,98 @@ end
             computed_density[point] = NaN
             neighbor_count[point] = 0
 
-            foreach(cache) do field
-                if field isa AbstractVector
-                    field[point] = NaN
-                else
-                    field[:, point] .= NaN
-                end
-            end
+            set_nan!(cache, point, ref_system)
```

Comment:
Does this work on the GPU?
```suggestion
            foreach(cache) do field
                field[.., point] .= NaN
            end
```
with `using EllipsisNotation`.

If not, this is still shorter and avoids future maintenance when systems are changed.
```suggestion
            foreach(cache) do field
                set_nan!(field, point)
            end
```
and
```julia
set_nan!(field::AbstractVector, i) = field[i] = NaN
set_nan!(field::AbstractMatrix, i) = field[:, i] .= NaN
```